chore(editor): Improve VS Code extension dev experience

connorshea opened the pull request:
Improve the experience of launching the VS Code extension for development purposes. Now it will build it for you automatically when launching the editor extension, rather than you needing to build it yourself.

We could also maybe keep a launch setup that doesn't do this alongside this one, if others prefer using that workflow.

See this video for the workflow of building+launching this from the debug menu in VS Code:

https://github.com/user-attachments/assets/738368d4-88fd-424b-ab96-4b1cfed38491

Sysix reviewed (changes requested):
The goal for the status bar is to manage both servers in the future in https://github.com/oxc-project/oxc/pull/15837:
<img width="227" height="283" alt="grafik" src="https://github.com/user-attachments/assets/e3ae7256-a0fb-440d-88a2-b294b590d1f9" />

There will be two Output Channels for each server :) 

----

Can you just improve the tooling please :)

connorshea commented:
Fair, rebased and removed the output channel thing! I think we may want to consider adding a command to trigger when clicking the status bar item, but can discuss that separately/later.

Sysix commented:
### Merge activity

* **Nov 26, 1:12 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Nov 26, 1:13 PM UTC**: `Sysix` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

connorshea committed:
chore(editor): Improve VS Code extension dev experience (#16087)

Improve the experience of launching the VS Code extension for development purposes. Now it will build it for you automatically when launching the editor extension, rather than you needing to build it yourself.

We could also maybe keep a launch setup that doesn't do this alongside this one, if others prefer using that workflow.

See this video for the workflow of building+launching this from the debug menu in VS Code:

https://github.com/user-attachments/assets/738368d4-88fd-424b-ab96-4b1cfed38491
